### PR TITLE
[WIP][1L] updated cleanup edomain templates script.

### PR DESCRIPTION
**Pull request by amavinag**
Purpose or Intent
=================
  * updated cleanup methods from template_upload scripts are reused in cleanup edomain templates
    script.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 67c1fc5259cb48fffc2752c407e87fdbe6874213:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: scripts/cleanup_edomain_templates.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def is_ovirt_engine_running(rhevm_ip, sshname, sshpass):
         return False
 
 
-def change_edomain_state(api, state, edomain):
+def change_edomain_state(api, state, edomain, provider):
```
Comment (psav) [inline scripts/cleanup_edomain_templates.py]: Is ```provider``` just a string?
Is api from mgmtsystem? What about passing in the provider object where we can get both api and the name from in one object?

File: scripts/cleanup_edomain_templates.py [hunk 2]
```diff
@@ -100,25 +102,35 @@ def get_edomain_path(api, edomain):
             edomain_conn.get_address())
 
 
-def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass, provider_ip, provider):
```
Comment (psav) [inline scripts/cleanup_edomain_templates.py]: Again here, the provider object should have the IP as well as the name

File: scripts/cleanup_edomain_templates.py [hunk 3]
```diff
@@ -100,25 +102,35 @@ def get_edomain_path(api, edomain):
...
     try:
-        print("RHEVM: Deleting the empty directories on edomain/vms file...")
-        ssh_client = make_ssh_client(edomainip, sshname, sshpass)
-        command = 'cd {}/master/vms && find . -maxdepth 1 -type d -empty -delete'.format(path)
+        ssh_client = make_ssh_client(provider_ip, sshname, sshpass)
+        edomain_path = edomainip + ':' + path
+        command = 'mkdir -p ~/tmp_filemount && mount -O tcp {} ~/tmp_filemount &&'.format(
+            edomain_path)
+        command += 'cd ~/tmp_filemount/master/vms &&'
+        command += 'find . -maxdepth 1 -type d -empty -delete &&'
```
Comment (RonnyPfannschmidt) [inline scripts/cleanup_edomain_templates.py]: wouldnt passing the explicit target path remove the need for the cd dance?

Comment (amavinag) [inline scripts/cleanup_edomain_templates.py]: @RonnyPfannschmidt agree :) will change that.

File: scripts/cleanup_edomain_templates.py (hunk 1 above)
Comment (amavinag) [inline scripts/cleanup_edomain_templates.py]: @psav that was just a string. In the new commit changed that to pass the provider_obj and fetch the api, provider_name from that.

File: scripts/cleanup_edomain_templates.py (hunk 2 above)
Comment (amavinag) [inline scripts/cleanup_edomain_templates.py]: @psav  In the new commit changed that to pass the provider_obj and fetch the provider_ip and provider_name from that.

File: scripts/cleanup_edomain_templates.py (hunk 3 above)
Comment (amavinag) [inline scripts/cleanup_edomain_templates.py]: @RonnyPfannschmidt changes updated with new commit.

File: scripts/cleanup_edomain_templates.py [hunk 4]
```diff
@@ -100,25 +105,37 @@ def get_edomain_path(api, edomain):
...
     """Cleanup all the empty directories on the edomain/edomain_id/master/vms
     else api calls will result in 400 Error with ovf not found,
-
     Args:
         path: path for vms directory on edomain.
         edomain: Export domain of chosen RHEVM provider.
         edomainip: edomainip to connect through ssh.
         sshname: edomain ssh credentials.
         sshpass: edomain ssh credentials.
+        provider: provider under execution
```
Comment (RonnyPfannschmidt) [inline scripts/cleanup_edomain_templates.py]: wrong docs

**Comment by RonnyPfannschmidt** [comment]
the change looks good, thanks for the update, please make sure the docstring is in sync

**Comment by mshriver** [comment]
@dajohnso @psav @RonnyPfannschmidt

It would take some investigation/learning on my part but I could take over this branch and get it prepped and merged?

Unless one of you, familiar with it, wants to own it?
Thoughts @dajohnso ?

**Comment by mshriver** [comment]
Closed for PR #4113 4113

**Pull request closed without merging**